### DISCUSSION TITLE
Difficulty with a particular site (instacart.com)

### DISCUSSION THREAD

**Discussion by tblancher** [qutebrowser/qutebrowser#7819] [Q&A]
I'm not sure how to debug this, I've had this problem with [InstaCart](https://instacart.com/store) in qutebrowser since the first time I tried navigating to it.  I believe I have to be logged into InstaCart to get to this page (navigating to it while logged out presents the login page).

Here's my qutebrowser version (from `:version`):

```
qutebrowser v2.5.4
Git commit: 6aa53a3c9 on makepkg (2023-07-30 11:18:05 +1200)
Backend: QtWebEngine 6.5.2, based on Chromium 108.0.5359.220 (from api)
Qt: 6.5.2
```
This is on Arch Linux.  Here's the erorr page I get, it doesn't really say much:

![2023-08-05-132324_1203x760_scrot](https://github.com/qutebrowser/qutebrowser/assets/1856323/f9ce1a0a-edcc-46d0-80ab-ec80ed32cd34)

I tried loading this with `--temp-basedir`, and I get the same error from InstaCart, once I logged in.  The only error I see in the console was this: 

```
[449380:449380:0805/141307.214533:ERROR:interface_endpoint_client.cc(694)] Message 4 rejected by interface blink.mojom.WidgetHost
14:13:10 ERROR: JS: [userscript:_qute_js:462] Uncaught InvalidStateError: Failed to set the 'selectionStart' property on 'HTMLInputElement': The input element's type ('email') does not support selection.
```
But I believe that was when I tried to use the 'f' hints to make the username/email address field active when I was logging in.

I didn't file a bug report since I don't know what I should gather to troubleshoot this.  The other browser I use for this is Vivaldi, which is based on a newer version of chromium, and that doesn't have the same problem.  I've only seen this in qutebrowser Qt 6 (I never logged into InstaCart in the Qt 5 version, but I figure that would be worse!).

EDIT:  I loaded it in `:devtools`, and it looks like there are a couple of JavaScript files which are blocked:  
![2023-08-05-142506_1583x513_scrot](https://github.com/qutebrowser/qutebrowser/assets/1856323/e81c584c-6053-4c22-ad9c-a68c947b72fb)
I don't typically block JS from running, so I don't know why these are blocked.  I'm going to check to see if I have any JS settings which might cause this.   I tried whitelisting the [instacart.com/*](https://instacart.com) domain in `content.blocking.whiltelist ["https://instacart.com/*"]`, but that didn't seem to help.

**Comment by The-Compiler** [comment]
I opened an account and I can reproduce. I tried with `--temp-basedir` and then (after solving multiple captchas) they now consistently hit me with "We couldn’t validate the security of this request. Sorry!" when trying to log in. So, no idea what's wrong, but I probably won't be able to do much about it...

**Reply by tblancher** [reply]
I was editing my original post when you commented,  It looks like some Javascript is being blocked, and that likely causes the site to throw an error.

**Reply by The-Compiler** [reply]
It's not that. With `set -u www.instacart.com content.blocking.enabled false`, it still happens (setting `.whitelist` applies to hosts where things are loaded from).

**Reply by tblancher** [reply]
It looks like _script.js_ is loaded from a third-party domain:
![2023-08-05-155001_1258x218_scrot](https://github.com/qutebrowser/qutebrowser/assets/1856323/a2f391e8-3272-492e-8d98-772eb2d92341)

It also looks like the Google Tag Manager ( _gtm.js_ ) is blocked as well:
![2023-08-05-155209_1250x169_scrot](https://github.com/qutebrowser/qutebrowser/assets/1856323/6a16a450-d751-4569-9568-3f10e6710b95)

How can I let InstaCart load script content from third-party domains?  It sounds like allowing it globally might leave me open to malicious scripts if I'm not careful.

**Reply by tblancher** [reply]
A little bit of progress... I added the sources for _script.js_ and _gtm.js_ to `content.blocking.whitelist` and these two particular errors went away. Unfortunately, there were many other errors (it went from 9 errors in `:devtools` to 27, mostly with the same `net::ERR BLOCKED BY CLIENT` on several different Javescript files).

So I disabled adblock for the InstaCart URL pattern:  
```
:set -u "https://*.instacart.com/*" content.blocking.enabled false
```
And now I get none of the Javascript loading errors.  But the problem isn't fixed, there are 8 more errors (some of them were seen before):
![2023-08-05-162326_1038x337_scrot](https://github.com/qutebrowser/qutebrowser/assets/1856323/0fde960c-e16f-4e6d-a60b-17d50eece3e9)
![2023-08-05-162455_1045x649_scrot](https://github.com/qutebrowser/qutebrowser/assets/1856323/e8c38aec-a2be-4f29-8f55-1ffce809e288)
Any one of these could be the culprit.

By the way, thanks for taking the time to look into this!

**Reply by tblancher** [reply]
I loaded devtools in Vivaldi, and I see quite a few third-party Javascript files blocked, so that likely isn't it.  I don't see any of the same errors in Vivaldi, I'm wondering if the _qute_stylesheet_ userscript is at fault.  I can't find a file with this name on my filesystem.

That `net::ERR NAME NOT RESOLVED` looked to be temporary, reloading the page doesn't show this error anymore.  So it's either the invalid time values, or these inline executions related to stylesheets, as best as I can tell.
